.github/workflows/publish_all_packages.yml

```diff
@@ -5,23 +5,12 @@ on:
       - master
 
 jobs:
-  publish_all_packages:
+  publish:
+    permissions:
+      id-token: write # Required for authentication using OIDC
     runs-on: ubuntu-latest
-    container:
-      image: google/dart:latest
     steps:
-      - uses: actions/checkout@v1
-      - name: Setup credentials
-        run: |
-          mkdir -p ~/.pub-cache 
-          cat <<EOF > ~/.pub-cache/credentials.json
-          {
-            "accessToken":"${{ secrets.PUB_DEV_ACCESS_TOKEN }}",
-            "refreshToken":"${{ secrets.PUB_DEV_REFRESH_TOKEN }}",
-            "tokenEndpoint":"https://accounts.google.com/o/oauth2/token",
-            "scopes": [ "openid", "https://www.googleapis.com/auth/userinfo.email" ],
-            "expiration": 1570721159347
-          }
-          EOF
-      - name: Publish package
+      - uses: actions/checkout@v3
+      - uses: dart-lang/setup-dart@v1
```

Comment:
We should fix these actions to a given commit-hash. it's a bit more secure.
For future us 😄